packages/core/lib/package.js

```diff
@@ -1,351 +1,214 @@
-                  // Bail early if we foun done.
-                  async.each(
-                    deployed_networks,
-                    function(deployed_network, finished) {
-                      Networks.matchesNetwork(
-                        network_id,
-                        options.networks[deployed_network],
-                        function(err, matches) {
-                          if (err) return finished(err);
-                          if (matches) {
-                            var uri = uris[deployed_network];
-
-                            if (!deployments[uri]) {
-                              deployments[uri] = {};
-                            }
-
-                            deployments[uri][data.contractName] = {
-                              contract_type: data.contractName, // TODO: Handle conflict resolution
-                              address: data.networks[network_id].address
-                            };
-
-                            return finished("bail early");
-                          }
-                          finished();
-                        }
-                      );
-                    },
-                    function(err) {
-                      if (err && err !== "bail early") {
-                        return reject(err);
-                      }
+// Returns a list of releasable artifacts
+// aka contract_types and deployments found in all artifacts
+function releasable_artifacts(options) {
+  var files = fs.readdirSync(options.contracts_build_directory);
+  files = files.filter(file => file.includes(".json"));
 
-                      accept();
-                    }
-                  );
-                })
-              );
-            });
-          });
+  if (!files.length) {
+    var msg =
+      "Could not locate any releasable artifacts in " +
+      options.contracts_build_directory +
+      ". " +
+      "Run `truffle compile` before releasing.";
 
-          return Promise.all(matching_promises);
-        })
-        .then(function() {
-          var to_return = {
-            contract_types: contract_types,
-            deployments: deployments
-          };
+    return new Error(msg);
+  }
 
-          callback(null, to_return);
-        })
-        .catch(callback);
-    });
+  const fileData = [];
+  const blockchainUri = options.networks.mainnet.network_id;
+  for (let file of files) {
+    const fileContents = JSON.parse(
+      fs.readFileSync(
+        path.join(options.contracts_build_directory, file),
+        "utf8"
+      )
+    );
+    // hacky af - blockchain uri wasn't getting picked up as network_id for 2 of 3 migrations
+    // some artifacts are coming through w/ network ids that are ints
+    // the blockchain uri coming through is what is set as the network_id
+    // not a freshly created one with latest block via BlockchainUtils
+    const foundNetworkId = Object.keys(fileContents.networks)[0];
+    if (foundNetworkId !== blockchainUri) {
```

Comment:
note that in order to construct a blockchain URI, you're going to need to actually connect to the networks in the truffle-config, enumerating over each defined network, and for each of these networks, find the relevant block info

sorry